codegenerator/cli/templates/static/codegen/src/eventFetching/chainWorkers/HyperFuelWorker.res

```diff
@@ -0,0 +1,352 @@
+    await waitForNextBlockBeforeQuery(
+      ~serverUrl=T.endpointUrl,
+      ~fromBlock,
+      ~currentBlockHeight,
+      ~setCurrentBlockHeight,
+      ~logger,
+    )
+
+    //Instantiate each time to add new registered contract addresses
+    let contractsReceiptQuery = T.contracts->Belt.Array.keepMap((contract): option<
+      HyperFuel.contractReceiptQuery,
+    > => {
+      switch contractAddressMapping->ContractAddressingMap.getAddressesFromContractName(
+        ~contractName=contract.name,
+      ) {
+      | [] => None
+      | addresses => Some({
+          addresses,
+          logIds: contract.events->Js.Array2.map(eventMod => {
+            let module(Event: Types.Event) = eventMod
+            Event.topic0
+          })
+        })
+      }
+    })
+
+    let startFetchingBatchTimeRef = Hrtime.makeTimer()
+
+    //fetch batch
+    let pageUnsafe = await Helpers.queryLogsPageWithBackoff(
+      () =>
+        HyperFuel.queryLogsPage(
+          ~serverUrl=T.endpointUrl,
+          ~fromBlock,
+          ~toBlock,
+          ~contractsReceiptQuery,
+        ),
+      logger,
+    )
+
+    let pageFetchTime =
+      startFetchingBatchTimeRef->Hrtime.timeSince->Hrtime.toMillis->Hrtime.intFromMillis
+
+    {page: pageUnsafe, pageFetchTime}
+  }
+
+  let fetchBlockRange = async (
+    ~query: blockRangeFetchArgs,
+    ~logger,
+    ~currentBlockHeight,
+    ~setCurrentBlockHeight,
+  ) => {
+    let config = Config.getGenerated()
+    let mkLogAndRaise = ErrorHandling.mkLogAndRaise(~logger, ...)
+    try {
+      let {fetchStateRegisterId, partitionId, fromBlock, contractAddressMapping, toBlock} = query
+      let startFetchingBatchTimeRef = Hrtime.makeTimer()
+      //fetch batch
+      let {page: pageUnsafe, pageFetchTime} = await getNextPage(
+        ~fromBlock,
+        ~toBlock,
+        ~currentBlockHeight,
+        ~contractAddressMapping,
+        ~logger,
+        ~setCurrentBlockHeight,
+      )
+
+      //set height and next from block
+      let currentBlockHeight = pageUnsafe.archiveHeight
+
+      logger->Logging.childTrace({
+        "message": "Retrieved event page from server",
+        "fromBlock": fromBlock,
+        "toBlock": pageUnsafe.nextBlock - 1,
+      })
+
+      //The heighest (biggest) blocknumber that was accounted for in
+      //Our query. Not necessarily the blocknumber of the last log returned
+      //In the query
+      let heighestBlockQueried = pageUnsafe.nextBlock - 1
+
+      let lastBlockQueriedPromise: promise<
+        ReorgDetection.blockData,
+      > = // switch pageUnsafe.rollbackGuard {
+      // //In the case a rollbackGuard is returned (this only happens at the head for unconfirmed blocks)
+      // //use these values
+      // | Some({blockNumber, timestamp, hash}) =>
+      //   {
+      //     ReorgDetection.blockNumber,
+      //     blockTimestamp: timestamp,
+      //     blockHash: hash,
+      //   }->Promise.resolve
+      // | None =>
```

Comment:
I'm not actually going to look at this file since you say it's not currently working. Just noting there's quite a bit of commented code.